mlflow/store/local_artifact_repo.py

```diff
@@ -39,6 +39,27 @@ def log_artifacts(self, local_dir, artifact_path=None):
             mkdir(artifact_dir)
         dir_util.copy_tree(src=local_dir, dst=artifact_dir)
 
+    def download_artifacts(self, artifact_path, dst_path=None):
+        """
+        Download an artifact file or directory to a local directory if applicable, and return a
+        local path for it.
+        The caller is responsible for managing the lifecycle of the downloaded artifacts. If
+        dst_path is unspecified, the downloaded artifact may not be mutated in general.
```

Comment:
To me, `mutated` implies that the artifact file / artifact directory contents will be somehow altered by `download_artifacts()`, which is not the case. Perhaps something like the following:

```
Artifacts tracked by ``LocalArtifactRepository`` already exist on the local filesystem. If ``dst_path`` is ``None``, the absolute filesystem path of the specified artifact is returned. If ``dst_path`` is not ``None``, the local artifact is copied to ``dst_path``.
```